functional-test/src/main/java/org/zanata/page/WebDriverFactory.java

```diff
@@ -136,6 +144,29 @@ private WebDriver configureChromeDriver() {
                                 capabilities)));
     }
 
+    /**
+     * Returns a File pointing to a chromedriver binary, searching the PATH
+     * if necessary. If the property 'webdriver.chrome.driver' points to an
+     * executable, uses that.  Otherwise, searches PATH for the specified
+     * executable.  Searches for 'chromedriver' if the property is null or
+     * empty.
+     * @return a File pointing to the executable
+     * @deprecated use ChromeDriverService.createDefaultService() if you can
+     */
+    @Deprecated
+    private File getChromeDriver() {
+        String exeName = PropertiesHolder.getProperty("webdriver.chrome.driver");
+        if (exeName == null || exeName.isEmpty()) {
```

Comment:
Yes, but then it's less clear (to me and to a static null checker) that
exeName is non-null in the `else if` below.

On 2014-12-22 14:09, Damian Jansen wrote:
> In functional-test/src/main/java/org/zanata/page/WebDriverFactory.java
> <https://github.com/zanata/zanata-server/pull/655#discussion-diff-22154042>:
> 
>> @@ -136,6 +144,29 @@ private WebDriver configureChromeDriver() {
>>                                  capabilities)));
>>      }
>>  
>> +    /**
>> +     * Returns a File pointing to a chromedriver binary, searching the PATH
>> +     * if necessary. If the property 'webdriver.chrome.driver' points to an
>> +     * executable, uses that.  Otherwise, searches PATH for the specified
>> +     * executable.  Searches for 'chromedriver' if the property is null or
>> +     * empty.
>> +     * @return a File pointing to the executable
>> +     * @deprecated use ChromeDriverService.createDefaultService() if you can
>> +     */
>> +    @Deprecated
>> +    private File getChromeDriver() {
>> +        String exeName = PropertiesHolder.getProperty("webdriver.chrome.driver");
>> +        if (exeName == null || exeName.isEmpty()) {
> 
>     if (StringUtils.isEmpty(exeName)) {
> 
> is an alternative.
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/655/files#r22154042>.
> 


-- 
Sean Flanigan

Senior Software Engineer
Engineering - Internationalisation
Red Hat